client/dogs/FormPage.js

```diff
@@ -296,6 +297,7 @@ export type InitialProps = {|
 
 export type Props = {
   ...InitialProps,
+  children: Element<any>,
```

Comment:
It is not needed. This was a rabbit hole I went down trying to fix flow errors. I had initially add `import React, { Element } from 'react';` to fix a parameter error and then flow wouldn't let me declare `Element` without using it, so I added that. I've reverted both of those changes since they aren't necessary with the refactored code. Thanks!